testsuite/src/main/java/io/netty/testsuite/transport/socket/DatagramUnicastTest.java

```diff
@@ -117,12 +121,97 @@ public void testSimpleSendWithoutBind(Bootstrap sb, Bootstrap cb) throws Throwab
     }
 
+    @Test
+    public void testSimpleSendWithConnect() throws Throwable {
+        run();
+    }
+
+    public void testSimpleSendWithConnect(Bootstrap sb, Bootstrap cb) throws Throwable {
+        testSimpleSendWithConnect0(sb, cb, Unpooled.directBuffer().writeBytes(BYTES), BYTES, 1);
+        testSimpleSendWithConnect0(sb, cb, Unpooled.directBuffer().writeBytes(BYTES), BYTES, 4);
+    }
+
     @SuppressWarnings("deprecation")
     private void testSimpleSend0(Bootstrap sb, Bootstrap cb, ByteBuf buf, boolean bindClient,
                                  final byte[] bytes, int count)
             throws Throwable {
+        cb.handler(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            public void channelRead0(ChannelHandlerContext ctx, Object msgs) throws Exception {
+                // Nothing will be sent.
+            }
+        });
+
+        final CountDownLatch latch = new CountDownLatch(count);
+        Channel sc = setupServerChannel(sb, bytes, count, latch);
+
+        Channel cc;
+        if (bindClient) {
+            cc = cb.bind().sync().channel();
+        } else {
+            cb.option(ChannelOption.DATAGRAM_CHANNEL_ACTIVE_ON_REGISTRATION, true);
+            cc = cb.register().sync().channel();
+        }
+
+        for (int i = 0; i < count; i++) {
+            cc.write(new DatagramPacket(buf.retain().duplicate(), addr));
+        }
+        // release as we used buf.retain() before
+        buf.release();
+        cc.flush();
+        assertTrue(latch.await(10, TimeUnit.SECONDS));
+
+        sc.close().sync();
+        cc.close().sync();
+    }
+
+    private void testSimpleSendWithConnect0(Bootstrap sb, Bootstrap cb, ByteBuf buf, final byte[] bytes, int count)
+            throws Throwable {
+        cb.handler(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            public void channelRead0(ChannelHandlerContext ctx, Object msgs) throws Exception {
+                // Nothing will be sent.
+            }
+        });
+
         final CountDownLatch latch = new CountDownLatch(count);
+        Channel sc = setupServerChannel(sb, bytes, count, latch);
+        DatagramChannel cc = null;
+        try {
+            cc = (DatagramChannel) cb.connect(sc.localAddress()).sync().channel();
+
+            for (int i = 0; i < count; i++) {
+                cc.write(buf.retain().duplicate());
+            }
+            cc.flush();
+            assertTrue(latch.await(10, TimeUnit.SECONDS));
+
+            assertTrue(cc.isConnected());
+
+            // Test what happens when we call disconnect()
+            cc.disconnect().syncUninterruptibly();
+            assertFalse(cc.isConnected());
 
+            ChannelFuture future = cc.writeAndFlush(
+                    buf.retain().duplicate()).awaitUninterruptibly();
+            if (!(cc instanceof OioDatagramChannel)) {
+                // TODO: Also handle OIO
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)